Spoke too fast. @iguinn I get this:
```
make[1]: Entering directory `/user/legend-swdev-scripts/mage-post-proc/Apps'
g++ -O2 -g -O2 -fPIC -g -O2 -Wextra -Wall -DMGWF_HAS_FFTW3 -DCUSTOMIZE_STREAM -std=c++11 -I"/opt/clhep/include"   -I/opt/root/include -pthread -std=c++11 -m64  -I/user/.local/include/mgdo -I/user/.local/include/tam -I../BaseClasses -I../Processing -I../SiggenWrapper   -o process_basic process_basic.o   -L/user/.local/lib -lMGDOBase -lMGDOTransforms -lMGDOGerdaTransforms -lMGDORoot -lMGDOGerda -lTAM -lMGDOTabree -lMGDOCrystalGeometry -L../lib -lMPPBaseClasses -lMPPProcessing -lMPPSiggenWrapper -L/opt/root/lib -lCore -lImt -lRIO -lNet -lHist -lGraf -lGraf3d -lGpad -lROOTVecOps -lTree -lTreePlayer -lRint -lPostscript -lMatrix -lPhysics -lMathCore -lThread -lMultiProc -lROOTDataFrame -pthread -lm -ldl -rdynamic -lTreePlayer -lMathMore  -lfftw3  -L"/opt/clhep/lib" -lCLHEP-2.4.1.0
process_basic.o: In function `MPPSensVolData<void>::~MPPSensVolData()':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPSensVolData.hh:17: undefined reference to `vtable for MPPSensVolData<void>'
process_basic.o: In function `~MPPWindowedData':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
process_basic.o: In function `MPPSensVolData<void>::~MPPSensVolData()':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPSensVolData.hh:17: undefined reference to `vtable for MPPSensVolData<void>'
process_basic.o: In function `~MPPWindowedData':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
process_basic.o: In function `MPPSensVolData<void>::IsA() const':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPSensVolData.hh:60: undefined reference to `MPPSensVolData<void>::Class()'
process_basic.o: In function `MPPSensVolData<void>::ShowMembers(TMemberInspector&) const':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPSensVolData.hh:60: undefined reference to `MPPSensVolData<void>::Class()'
process_basic.o: In function `~MPPWindowedData':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
process_basic.o: In function `MPPWindowedData<double>::assign(MPPWindowedData<void> const*)':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:92: undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o: In function `~MPPWindowedData':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: undefined reference to `vtable for MPPWindowedData<void>'
process_basic.o:/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:23: more undefined references to `vtable for MPPWindowedData<void>' follow
process_basic.o: In function `MPPWindowedData<bool>::assign(MPPWindowedData<void> const*)':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:204: undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o: In function `MPPWindowedData<unsigned char>::assign(MPPWindowedData<void> const*)':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:92: undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o: In function `MPPWindowedData<std::vector<std::vector<MGTMCStepData const*, std::allocator<MGTMCStepData const*> >, std::allocator<std::vector<MGTMCStepData const*, std::allocator<MGTMCStepData const*> > > > >::assign(MPPWindowedData<void> const*)':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:92: undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o: In function `MPPWindowedData<std::vector<int, std::allocator<int> > >::assign(MPPWindowedData<void> const*)':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:92: undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o: In function `MPPWindowedData<std::vector<unsigned char, std::allocator<unsigned char> > >::assign(MPPWindowedData<void> const*)':
/user/legend-swdev-scripts/mage-post-proc/Apps/../BaseClasses/MPPWindowedData.hh:92: undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o:(.data.rel.ro._ZTI15MPPWindowedDataIbE[_ZTI15MPPWindowedDataIbE]+0x18): more undefined references to `typeinfo for MPPWindowedData<void>' follow
process_basic.o:(.data.rel.ro._ZTI14MPPSensVolDataIbE[_ZTI14MPPSensVolDataIbE]+0x18): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTI14MPPSensVolDataIhE[_ZTI14MPPSensVolDataIhE]+0x18): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTI15MPPWindowedDataISt6vectorIiSaIiEEE[_ZTI15MPPWindowedDataISt6vectorIiSaIiEEE]+0x18): undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o:(.data.rel.ro._ZTI14MPPSensVolDataIiE[_ZTI14MPPSensVolDataIiE]+0x18): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTI15MPPWindowedDataISt6vectorIS0_IPK13MGTMCStepDataSaIS3_EESaIS5_EEE[_ZTI15MPPWindowedDataISt6vectorIS0_IPK13MGTMCStepDataSaIS3_EESaIS5_EEE]+0x18): undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o:(.data.rel.ro._ZTI14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE[_ZTI14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE]+0x18): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTI15MPPWindowedDataIdE[_ZTI15MPPWindowedDataIdE]+0x18): undefined reference to `typeinfo for MPPWindowedData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIbE64_S_IvE[_ZTV14MPPSensVolDataIbE]+0x10): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIbE64_S_IvE[_ZTV14MPPSensVolDataIbE]+0xa0): undefined reference to `MPPSensVolData<void>::Streamer(TBuffer&)'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIbE64_S_IvE[_ZTV14MPPSensVolDataIbE]+0x318): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIbE64_S_IvE[_ZTV14MPPSensVolDataIbE]+0x4f0): undefined reference to `virtual thunk to MPPSensVolData<void>::Streamer(TBuffer&)'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIhE96_S_IvE[_ZTV14MPPSensVolDataIhE]+0x10): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIhE96_S_IvE[_ZTV14MPPSensVolDataIhE]+0xa0): undefined reference to `MPPSensVolData<void>::Streamer(TBuffer&)'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIhE96_S_IvE[_ZTV14MPPSensVolDataIhE]+0x318): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIhE96_S_IvE[_ZTV14MPPSensVolDataIhE]+0x4f0): undefined reference to `virtual thunk to MPPSensVolData<void>::Streamer(TBuffer&)'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIiE96_S_IvE[_ZTV14MPPSensVolDataIiE]+0x10): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIiE96_S_IvE[_ZTV14MPPSensVolDataIiE]+0xa0): undefined reference to `MPPSensVolData<void>::Streamer(TBuffer&)'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIiE96_S_IvE[_ZTV14MPPSensVolDataIiE]+0x318): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataIiE96_S_IvE[_ZTV14MPPSensVolDataIiE]+0x4f0): undefined reference to `virtual thunk to MPPSensVolData<void>::Streamer(TBuffer&)'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE96_S_IvE[_ZTV14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE]+0x10): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE96_S_IvE[_ZTV14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE]+0xa0): undefined reference to `MPPSensVolData<void>::Streamer(TBuffer&)'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE96_S_IvE[_ZTV14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE]+0x318): undefined reference to `typeinfo for MPPSensVolData<void>'
process_basic.o:(.data.rel.ro._ZTC14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE96_S_IvE[_ZTV14MPPSensVolDataISt6vectorIPK13MGTMCStepDataSaIS3_EEE]+0x4f0): undefined reference to `virtual thunk to MPPSensVolData<void>::Streamer(TBuffer&)'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `MPPWaveformIDData::Class()'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `ROOT::GenerateInitInstance(MPPEventIDHeader const*)'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `MPPVersionInfo::Streamer(TBuffer&)'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `ROOT::GenerateInitInstance(MPPWaveformIDData const*)'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `MPPEventIDHeader::Class()'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `MPPEventIDHeader::Streamer(TBuffer&)'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `MPPClusterData::Class()'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `ROOT::GenerateInitInstance(MPPVersionInfo const*)'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `MPPVersionInfo::Class()'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `ROOT::GenerateInitInstance(MPPClusterData const*)'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `MPPClusterData::Streamer(TBuffer&)'
/user/.local/lib/libMPPBaseClasses.so: undefined reference to `MPPWaveformIDData::Streamer(TBuffer&)'
collect2: error: ld returned 1 exit status
make[1]: *** [process_basic] Error 1
make[1]: Leaving directory `/user/legend-swdev-scripts/mage-post-proc/Apps'
make: *** [install] Error 2
```
I'm compiling in `legend-base:20210207212256`.

--- a/installMaGe.py
+++ b/installMaGe.py
@@ -124,26 +124,24 @@ except subprocess.CalledProcessError:
     
 # install MGDO
 os.chdir('MGDO')
-subprocess.run(f'./configure --prefix={install_path} --enable-streamers --enable-tam --enable-tabree'.split())
-subprocess.run(['make', f'-j{args.jobs}'])
-if args.jobs>1:
-    subprocess.run(['make', '-j1']) # make -j>1 doesn't always succeed
-subprocess.run(['make', 'install', '-j1'])
+subprocess.run(f'./configure --prefix={install_path} --enable-streamers --enable-tam --enable-tabree'.split(), check=True)
+subprocess.run(f'make svninfo static -j{args.jobs}'.split(), check=True)
+subprocess.run('make', check=True)
+subprocess.run('make install'.split(), check=True)
 os.chdir('..')
 
 # install MaGe
-subprocess.run(f'cmake -S MaGe/source -B MaGe/build -DCMAKE_INSTALL_PREFIX={install_path}'.split())
+subprocess.run(f'cmake -S MaGe/source -B MaGe/build -DCMAKE_INSTALL_PREFIX={install_path}'.split(), check=True)
 os.chdir('MaGe/build')
-subprocess.run(['make', f'-j{args.jobs}'])
-subprocess.run(['make', 'install', '-j1'])
+subprocess.run(f'make -j{args.jobs}'.split(), check=True)
+subprocess.run('make install'.split(), check=True)
 os.chdir('../..')
 
 # install mage-post-proc
 os.chdir('mage-post-proc')
-subprocess.run(['make', f'-j{args.jobs}'])
-if args.jobs>1:
-    subprocess.run(['make', '-j1']) # make -j>1 doesn't always succeed
-subprocess.run(['make', 'install', '-j1'])
+subprocess.run(f'make -j{args.jobs} gitrev siggen static'.split(), check=True)
+subprocess.run('make', check=True)
+subprocess.run('make install'.split(), check=True)
 os.chdir(original_pwd)
 
 print('Installation complete. If desired, add the following line to your login script.')

--- a/installMaGe.py
+++ b/installMaGe.py
@@ -57,6 +57,11 @@ os.makedirs(args.buildpath, exist_ok=True)
 os.chdir(args.buildpath)
 pwd = os.getcwd()
 
+# helper function to run command line commands. Print command, then run it, and raise errors on failure
+def cmd(command):
+    print(command)
+    subprocess.run(command.split(), check=True)
+
 # determine install_path 
 install_path = args.installpath if args.installpath else pwd
 if not os.path.isdir(install_path):
@@ -121,34 +126,34 @@ try:
     address = 'git@github.com:' if args.authentication=='ssh' else\
               'https://github.com/'
     if not os.path.exists('MGDO'):
-        subprocess.run(f'git clone -b {args.mgdobranch} {address}{args.mgdofork}/MGDO.git'.split(), check=True)
+        cmd(f'git clone -b {args.mgdobranch} {address}{args.mgdofork}/MGDO.git')
     if not os.path.exists('MaGe'):
-        subprocess.run(f'git clone -b {args.magebranch} {address}{args.magefork}/MaGe.git MaGe/source'.split(), check=True)
+        cmd(f'git clone -b {args.magebranch} {address}{args.magefork}/MaGe.git MaGe/source')
     if not os.path.exists('mage-post-proc'):
-        subprocess.run(f'git clone -b {args.mppbranch} {address}{args.mppfork}/mage-post-proc.git'.split(), check=True)
+        cmd(f'git clone -b {args.mppbranch} {address}{args.mppfork}/mage-post-proc.git')
 except subprocess.CalledProcessError:
     sys.exit()
     
 # install MGDO
 os.chdir('MGDO')
-subprocess.run(f'./configure --prefix={install_path} --enable-streamers --enable-tam --enable-tabree'.split(), check=True)
-subprocess.run(f'make svninfo static -j{args.jobs}'.split(), check=True)
-subprocess.run('make', check=True)
-subprocess.run('make install'.split(), check=True)
+cmd(f'./configure --prefix={install_path} --enable-streamers --enable-tam --enable-tabree')
+cmd(f'make svninfo static -j{args.jobs}')
+cmd('make')
+cmd('make install')
 os.chdir('..')
 
 # install MaGe
-subprocess.run(f'cmake -S MaGe/source -B MaGe/build -DCMAKE_INSTALL_PREFIX={install_path}'.split(), check=True)
+cmd(f'cmake -S MaGe/source -B MaGe/build -DCMAKE_INSTALL_PREFIX={install_path}')
 os.chdir('MaGe/build')
-subprocess.run(f'make -j{args.jobs}'.split(), check=True)
-subprocess.run('make install'.split(), check=True)
+cmd(f'make -j{args.jobs}')
+cmd('make install')
 os.chdir('../..')
 
 # install mage-post-proc
 os.chdir('mage-post-proc')
-subprocess.run(f'make -j{args.jobs} gitrev siggen static'.split(), check=True)
-subprocess.run('make', check=True)
-subprocess.run('make install'.split(), check=True)
+cmd(f'make -j{args.jobs} gitrev siggen static')
+cmd('make')
+cmd('make install')
 os.chdir(original_pwd)
 
 print('Installation complete. If desired, add the following line to your login script.')

--- a/installMaGe.py
+++ b/installMaGe.py
@@ -30,6 +30,11 @@ parser.add_argument('-b', '--buildpath', type=str, default='.',
                     help="Base path to build software in")
 parser.add_argument('-i', '--installpath', type=str,
                     help="Base path to install software in. By default use the build path.")
+parser.add_argument('-a', '--authentication', type=str,
+                    choices=['ssh', 'https'], default='ssh',
+                    help="""github authentication method. Options:
+                        ssh: ssh-rsa key
+                        https: personal access token""")
 parser.add_argument('-j', '--jobs', type=int, default=1,
                     help="Number of threads to run with Make")
 
@@ -113,12 +118,14 @@ with open('setup_mage.sh', 'w') as file:
 
 # download (user will enter password)
 try:
+    address = 'git@github.com:' if args.authentication=='ssh' else\
+              'https://github.com/'
     if not os.path.exists('MGDO'):
-        subprocess.run(f'git clone -b {args.mgdobranch} https://github.com/{args.mgdofork}/MGDO.git'.split(), check=True)
+        subprocess.run(f'git clone -b {args.mgdobranch} {address}{args.mgdofork}/MGDO.git'.split(), check=True)
     if not os.path.exists('MaGe'):
-        subprocess.run(f'git clone -b {args.magebranch} https://github.com/{args.magefork}/MaGe.git MaGe/source'.split(), check=True)
+        subprocess.run(f'git clone -b {args.magebranch} {address}{args.magefork}/MaGe.git MaGe/source'.split(), check=True)
     if not os.path.exists('mage-post-proc'):
-        subprocess.run(f'git clone -b {args.mppbranch} https://github.com/{args.mppfork}/mage-post-proc.git'.split(), check=True)
+        subprocess.run(f'git clone -b {args.mppbranch} {address}{args.mppfork}/mage-post-proc.git'.split(), check=True)
 except subprocess.CalledProcessError:
     sys.exit()
     

